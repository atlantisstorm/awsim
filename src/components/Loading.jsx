import React from 'react';
import loadingGraphic from "../graphics/loading.gif";
const Loading = () => {
  return (
    <div id="loading">
      <img src={loadingGraphic} />
    </div>
  )
}

export default Loading;