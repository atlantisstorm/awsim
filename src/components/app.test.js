import React from 'react';
import { render } from "@testing-library/react";
import App from './App';

// Important! jest, and testing in general, really doesn't seem to like import of loading.gif
// in 'loading.jsx' component resulting in various hard to resolve errors.  So to get around
// this I've added this little hack. Not ideal, but needs must when the devil drives!
import * as loading from "./Loading";
loading.default = jest.fn(() => { 
  return (
    <p>loading</p>
  )
});

describe('App component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});