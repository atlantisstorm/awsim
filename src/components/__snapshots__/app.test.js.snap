// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="main-wrapper"
  >
    <div
      id="loaded"
    >
      <div
        aria-placeholder="Your instances will be here in a instant!:p"
        data-testid="instances"
        id="instances"
      >
        <table
          class="roundedTable"
        >
          <thead>
            <tr>
              <th
                class="{fieldName} {extra_class}"
                data-testid="header-sort"
              >
                <div
                  class="title"
                >
                  <div
                    class="left"
                  >
                    Name 
                  </div>
                  <div
                    class="right"
                  >
                    â–²
                  </div>
                </div>
              </th>
              <th
                class="{fieldName} {extra_class}"
                data-testid="header-sort"
              >
                <div
                  class="title"
                >
                  <div
                    class="left"
                  >
                    Environment 
                  </div>
                  <div
                    class="right"
                  />
                </div>
              </th>
              <th
                class="{fieldName} {extra_class}"
                data-testid="header-sort"
              >
                <div
                  class="title"
                >
                  <div
                    class="left"
                  >
                    State Name 
                  </div>
                  <div
                    class="right"
                  />
                </div>
              </th>
              <th
                class="{fieldName} {extra_class}"
                data-testid="header-sort"
              >
                <div
                  class="title"
                >
                  <div
                    class="left"
                  >
                    Public Ip Address 
                  </div>
                  <div
                    class="right"
                  />
                </div>
              </th>
              <th
                class="{fieldName} {extra_class}"
                data-testid="header-sort"
              >
                <div
                  class="title"
                >
                  <div
                    class="left"
                  >
                    Private Ip Address 
                  </div>
                  <div
                    class="right"
                  />
                </div>
              </th>
              <th
                class="{fieldName} {extra_class}"
                data-testid="header-sort"
              >
                <div
                  class="title"
                >
                  <div
                    class="left"
                  >
                    Auto Scaling Group Name 
                  </div>
                  <div
                    class="right"
                  />
                </div>
              </th>
              <th
                class="{fieldName} {extra_class}"
                data-testid="header-sort"
              >
                <div
                  class="title"
                >
                  <div
                    class="left"
                  >
                    Platform 
                  </div>
                  <div
                    class="right"
                  />
                </div>
              </th>
              <th
                class="{fieldName} {extra_class}"
                data-testid="header-sort"
              >
                <div
                  class="title"
                >
                  <div
                    class="left"
                  >
                    Instance Id 
                  </div>
                  <div
                    class="right"
                  />
                </div>
              </th>
              <th
                class="{fieldName} {extra_class}"
                data-testid="header-sort"
              >
                <div
                  class="title"
                >
                  <div
                    class="left"
                  >
                    Image Id 
                  </div>
                  <div
                    class="right"
                  />
                </div>
              </th>
              <th
                class="{fieldName} {extra_class}"
                data-testid="header-sort"
              >
                <div
                  class="title"
                >
                  <div
                    class="left"
                  >
                    Instance Type 
                  </div>
                  <div
                    class="right"
                  />
                </div>
              </th>
              <th
                class="{fieldName} {extra_class}"
                data-testid="header-sort"
              >
                <div
                  class="title"
                >
                  <div
                    class="left"
                  >
                    Launch Time 
                  </div>
                  <div
                    class="right"
                  />
                </div>
              </th>
            </tr>
          </thead>
          <tbody />
          <tfoot>
            <tr>
              <td
                colspan="12"
              >
                0 instances found.
              </td>
            </tr>
          </tfoot>
        </table>
      </div>
    </div>
    <div
      id="actions"
    >
      <div
        id="filters"
      >
        <div
          class="filter"
        >
          <span
            class="label"
          >
            Name
          </span>
          <select
            data-testid="select-Name"
            id="Name"
            name="Name"
          >
            <option
              value=""
            />
            <option
              value="ftp"
            >
              ftp
            </option>
            <option
              value="image"
            >
              image
            </option>
            <option
              value="loghost"
            >
              loghost
            </option>
            <option
              value="minion"
            >
              minion
            </option>
            <option
              value="rx"
            >
              rx
            </option>
            <option
              value="sql"
            >
              sql
            </option>
            <option
              value="tx"
            >
              tx
            </option>
            <option
              value="workhorse"
            >
              workhorse
            </option>
          </select>
        </div>
        <div
          class="filter"
        >
          <span
            class="label"
          >
            Environment
          </span>
          <select
            data-testid="select-Environment"
            id="Environment"
            name="Environment"
          >
            <option
              value=""
            />
            <option
              value="dev"
            >
              dev
            </option>
            <option
              value="qa"
            >
              qa
            </option>
            <option
              value="prod"
            >
              prod
            </option>
          </select>
        </div>
        <div
          class="filter"
        >
          <span
            class="label"
          >
            State Name
          </span>
          <select
            data-testid="select-StateName"
            id="StateName"
            name="StateName"
          >
            <option
              value=""
            />
            <option
              value="running"
            >
              running
            </option>
            <option
              value="stopped"
            >
              stopped
            </option>
            <option
              value="terminated"
            >
              terminated
            </option>
          </select>
        </div>
        <div
          class="filter"
        >
          <span
            class="label"
          >
            Platform
          </span>
          <select
            data-testid="select-Platform"
            id="Platform"
            name="Platform"
          >
            <option
              value=""
            />
            <option
              value="windows"
            >
              windows
            </option>
          </select>
        </div>
        <div
          class="filter"
        >
          <span
            class="label"
          >
            Image Id
          </span>
          <select
            data-testid="select-ImageId"
            id="ImageId"
            name="ImageId"
          >
            <option
              value=""
            />
          </select>
        </div>
        <div
          class="filter"
        >
          <span
            class="label"
          >
            Instance Type
          </span>
          <select
            data-testid="select-InstanceType"
            id="InstanceType"
            name="InstanceType"
          >
            <option
              value=""
            />
          </select>
        </div>
        <div
          class="filter"
        >
          <span
            class="label"
          >
            Regions
          </span>
          <select
            data-testid="select-Regions"
            id="Regions"
            name="Regions"
          >
            <option
              value="eu-west-1"
            >
              eu-west-1
            </option>
            <option
              value="us-east-1"
            >
              us-east-1
            </option>
            <option
              value="us-west-1"
            >
              us-west-1
            </option>
            <option
              value="us-west-2"
            >
              us-west-2
            </option>
            <option
              value="ap-southeast-1"
            >
              ap-southeast-1
            </option>
            <option
              value="ap-southeast-2"
            >
              ap-southeast-2
            </option>
            <option
              value="ap-northeast-1"
            >
              ap-northeast-1
            </option>
            <option
              value="sa-east-1"
            >
              sa-east-1
            </option>
          </select>
        </div>
      </div>
      <div
        id="refresh-container"
      >
        <button
          data-testid="refresh-button"
          id="btnRefreshInstances"
          title="Refresh then apply selected filters"
        >
          Refresh Instances
        </button>
      </div>
      <div
        class="clear"
      />
    </div>
  </div>
</DocumentFragment>
`;
